### PR TITLE
dialects: (wasm) Add wat encoding

### PR DESCRIPTION
**Pull request by superlopuh**

### CODE REVIEW

File: xdsl/xdsl_opt_main.py [hunk 1]
```diff
@@ -209,6 +209,19 @@ def _output_riscv_asm(prog: ModuleOp, output: IO[str]):
...
             x86.ops.print_assembly(prog, output)
 
+        def _output_wat(prog: ModuleOp, output: IO[str]):
+            from xdsl.dialects.wasm import WasmModule
+            from xdsl.dialects.wasm.wat import WatPrinter
+
+            wasm_modules = tuple(op for op in prog.walk() if isinstance(op, WasmModule))
+            if len(wasm_modules) != 1:
+                print(f"invalid number of wasm modules ({len(wasm_modules)})")
+                return
```
Comment (superlopuh) [inline xdsl/xdsl_opt_main.py:219]: I'm not sure if we want this or just print in a loop

Comment (PapyChacal) [reply xdsl/xdsl_opt_main.py:219]: What would be a reason *not* to print in a loop?

Comment (superlopuh) [reply xdsl/xdsl_opt_main.py:219]: I can't think of any, I'll change it

**Review by Moxinilian — APPROVED** [review]
Super neat but your simple change makes my next PR even bigger 💀

**Pull request merged**